src/amo/components/AddonReviewCard/index.js

```diff
@@ -545,6 +548,22 @@ export class AddonReviewCardBase extends React.Component<InternalProps> {
       >
         <div className="AddonReviewCard-container">
           {errorHandler.renderErrorIfPresent()}
+          {/* Div to display Addon's name and icon */}
+          {this.props.isUserProfile &&
+            addonName &&
+            addonIconURL &&
+            !this.isReply() && (
+              <div className="AddonReviewCard-addonInfo">
+                <img src={addonIconURL} alt={`Icon for ${addonName}`} />
+                <span>
+                  <span className="AddonReviewCard-addonReviewLabel">
+                    Review of{' '}
+                  </span>
+                  <span className="AddonReviewCard-addonName">{addonName}</span>
```

Comment:
Icon and text need to be translated. See https://github.com/mozilla/addons-frontend/blob/master/docs/i18n.md#using-sprintf for more details on how to do this.